### ISSUE TITLE
Can't pass existing role_arn to target when create_bus and create_role is false

### ISSUE THREAD

**Issue by tzfromaz** [terraform-aws-modules/terraform-aws-eventbridge#41]
## Description

In using the `eventbridge_with_api_destination` module, I am trying to use an existing IAM role for my target with `create_bus=false` and `create_role=false`. I thought I could pass an existing role in to my target (`targets` being a map of objects with EventBridge Target definitions)  using `role_arn` but Terraform does not recognize this behavior.

terraform apply leads to:
```
 Error: Creating EventBridge Target failed: ValidationException: RoleArn is required for target arn:aws:events:us-east-1:<account#>:api-destination/test/f1d2a8de-0558-4927-a511-4b9dede00b4a.
│       status code: 400, request id: 371ce7a1-a725-40fe-b251-1161cf6a3d5b
│ 
│   with module.eventbridge_with_api_destination.aws_cloudwatch_event_target.this["test"],
│   on .terraform/modules/eventbridge_with_api_destination/main.tf line 60, in resource "aws_cloudwatch_event_target" "this":
│   60: resource "aws_cloudwatch_event_target" "this" {

```

## Versions

 - Terraform: 1.0.5

 - Provider(s): hashicorp/aws - 3.72.0

 - Module: eventbridge_with_api_destination - latest
 
## Reproduction

Use `eventbridge_with_api_destination` module and set create_bus=false and create_role=false. Create a target, rule, connection and api_destination. Try to pass an existing role's `role_arn` into the target object since TF is not creating a role and witness this error.

### Code Snippet to Reproduce

Used this target in the module:
```
targets = {
    test = [
      {
        name = "test"
        destination = "test-api"
        attach_role_arn = true (tried false here as well)
        role_arn = "arn:aws:iam::<account#>:role/test-role"
      }
    ]
```

## Expected behavior

All the resources including the `aws_cloudwatch_event_target` are created properly by passing in an existing role using `role_arn` within the targets section.

## Actual behavior

An error message stating that the role ARN is required to create a `aws_cloudwatch_event_target`.

**Comment by svenlito** [comment]
Hi, @tzfromaz thanks for the issue. 
Have you tried the following?

```
attach_role_arn = "arn:aws:iam::<account#>:role/test-role"
```

**Comment by tzfromaz** [comment]
> Hi, @tzfromaz thanks for the issue. Have you tried the following?
> 
> ```
> attach_role_arn = "arn:aws:iam::<account#>:role/test-role"
> ```

Thanks for reaching out @svenlito! Yes I just tried that with this code:
```
targets = {
    asg-datadog-events-TEST = [
      {
        name              = "datadog-api-destination-TEST"
        destination       = "datadog-api"
        attach_role_arn   = "arn:aws:iam::<account#>:role/test-role"
      }
    ]
  }
```
And am still getting (same as earlier error):
```
Error: Creating EventBridge Target failed: ValidationException: RoleArn is required for target
```

**Comment by antonbabenko** [comment]
There is a bug:
https://github.com/terraform-aws-modules/terraform-aws-eventbridge/blob/4ad24ec1ad1d1cb4ca3ed4022e49e48e1957565b/main.tf#L68

it should be updated to take into account the value of `each.value.attach_role_arn` if it is specified (not `null` or `false`).

**Comment by tzfromaz** [comment]
So is `attach_role_arn `supposed to be a boolean or arn string value? Elsewhere in `/examples` it is used as a boolean. @antonbabenko @svenlito

**Comment by antonbabenko** [comment]
This issue has been resolved in version 1.14.0 :tada:
